File: system-test/dotnet/SystemTests/Latest/AcceptanceTests/BrowseCouponsPositiveTest.cs
using SystemTests.Latest.AcceptanceTests.Base;
using Dsl.Core.Shop;
using Optivem.Testing;

namespace SystemTests.Latest.AcceptanceTests;

public class BrowseCouponsPositiveTest : BaseAcceptanceTest
{
    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task ShouldBeAbleToBrowseCoupons(Channel channel)
    {
        await Scenario(channel)
            .When().BrowseCoupons()
            .Then().ShouldSucceed();
    }

    [Theory]
    [ChannelData(ChannelType.API)]
    public async Task ShouldReturnPublishedCoupon(Channel channel)
    {
        await Scenario(channel)
            .Given().Coupon().WithCouponCode("BROWSE10").WithDiscountRate(0.1m)
            .When().BrowseCoupons()
            .Then().ShouldSucceed()
            .And().Coupons().ContainsCouponWithCode("BROWSE10");
    }
}

File: system-test/java/src/test/java/com/optivem/shop/systemtest/latest/acceptance/BrowseCouponsPositiveTest.java
package com.optivem.shop.systemtest.latest.acceptance;

import com.optivem.shop.systemtest.latest.acceptance.base.BaseAcceptanceTest;
import com.optivem.shop.testkit.channel.ChannelType;
import com.optivem.testing.Channel;
import org.junit.jupiter.api.TestTemplate;

class BrowseCouponsPositiveTest extends BaseAcceptanceTest {
    @TestTemplate
    @Channel({ChannelType.UI, ChannelType.API})
    void shouldBeAbleToBrowseCoupons() {
        scenario
                .when().browseCoupons()
                .then().shouldSucceed();
    }
}
